packages/muelu/research/regionMG/src/SetupRegionMatrix_def.hpp

```diff
@@ -167,41 +162,29 @@ void MakeQuasiregionMatrices(const RCP<Xpetra::CrsMatrixWrap<Scalar, LocalOrdina
                             *(rowImport),
                             Xpetra::INSERT);
 
-  quasiRegionMats->fillComplete();
-
   tm = Teuchos::null;
   tm = rcp(new TimeMonitor(*TimeMonitor::getNewTimer("MakeQuasiregionMatrices: 3 - Scale interface entries")));
 
   RCP<CrsMatrixWrap> quasiRegionCrsWrap = Teuchos::rcp_dynamic_cast<CrsMatrixWrap>(quasiRegionMats);
   RCP<CrsMatrix> quasiRegionCrs = quasiRegionCrsWrap->getCrsMatrix();
-  const LO numRows    = Teuchos::as<LO>(quasiRegionCrs->getNodeNumRows());
-  // const size_t numNNZ = quasiRegionCrs->getNodeNumEntries();
-
-  ArrayRCP<const size_t> rowPtrCopy;
-  ArrayRCP<const LO> colIndsCopy;
-  ArrayRCP<const SC> valuesCopy;
-  quasiRegionCrs->getAllValues(rowPtrCopy, colIndsCopy, valuesCopy);
-
-  *myOut << "Just extracted quasi region matrix data, numEntries="
-         << valuesCopy.size() << ", last entry ID: " << rowPtrCopy[numRows]
-         << ", colIndsCopy.size()=" << colIndsCopy.size()
-         << ", valuesCopy.size()=" << valuesCopy.size() << std::endl;
-
-  ArrayRCP<size_t> rowPtr(rowPtrCopy.size());
-  ArrayRCP<LO> colInds(colIndsCopy.size());
-  ArrayRCP<SC> values(valuesCopy.size());
-  rowPtr.deepCopy(rowPtrCopy.view(0, rowPtrCopy.size()));
-  colInds.deepCopy(colIndsCopy.view(0, colIndsCopy.size()));
-  values.deepCopy(valuesCopy.view(0, valuesCopy.size()));
+
+  Array<LO> tmp(regionMatVecLIDs());
+  std::sort(tmp.begin(), tmp.end());
+  auto vecEnd = std::unique(tmp.begin(), tmp.end());
+  auto vecStart = tmp.begin();
```

Comment:
`vecEnd` and `vecStart` are not super expressive variable names, but I don't have a better idea right now.